### PR TITLE
Fix(eos_validate_state): additional check in generate vars

### PR DESCRIPTION
**Pull request by ankudinov**
## Change Summary

`generate_vars_for_testing.j2` may fail when running `eos_validate_state` due to missing key in the dictionary. While the problem may be related to the repository, additional check must be implemented in the template to verify key existance.

## Related Issue(s)

none

## Component(s) name

`arista.avd.eos_validate_state`

## Proposed changes

Add additional test to verify key existance.

## How to test
The updated template was tested by executing validation in avd-quickstart environment and the repository with the problem.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_validate_state/templates/generate_vars_for_testing.j2 [hunk 1]
```diff
@@ -14,7 +14,10 @@ loopback0_reachability:
 {% set address = [] %}
 {% for node in groups[fabric_name] | arista.avd.natural_sort %}
 {%     if hostvars[node].vxlan_interface.Vxlan1.vxlan.source_interface is arista.avd.defined %}
-{%         do address.append(hostvars[node].loopback_interfaces[hostvars[node].vxlan_interface.Vxlan1.vxlan.source_interface].ip_address | ansible.netcommon.ipaddr('address')) %}
+{%         set vtep_reachability_int = hostvars[node].vxlan_interface.Vxlan1.vxlan.source_interface %}
+{%         if hostvars[node].loopback_interfaces[vtep_reachability_int].ip_address is arista.avd.defined and hostvars[node].loopback_interfaces.vtep_reachability_int.ip_address is not none %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_validate_state/templates/generate_vars_for_testing.j2]: May I know the reason to check this variable ```hostvars[node].loopback_interfaces.vtep_reachability_int.ip_address```

**Review by tgodaA — APPROVED** [review]
LGTM

**Review by xaviramon — APPROVED** [review]
LGTM

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Pull request merged**